### PR TITLE
fixes formula to comply to its spec

### PR DESCRIPTION
**Pull request by fortune-juggle**
The reworks in merge #110 broke this formulas kitchen based specs.
This MR changes a few minor things, mostly  pillar.example and other inconsistent documentations.
Also, the use of db_user for `postgres_tablespace.present` and `postgres_database.present` now fits the states options, as db_password et al can be specified as well.

Please see the specs output below.

```
-----> Running serverspec test suite
-----> Installing Serverspec..
Fetching: rspec-support-3.5.0.gem (100%)
Fetching: rspec-core-3.5.2.gem (100%)
Fetching: diff-lcs-1.2.5.gem (100%)
Fetching: rspec-expectations-3.5.0.gem (100%)
Fetching: rspec-mocks-3.5.0.gem (100%)
Fetching: rspec-3.5.0.gem (100%)
Fetching: rspec-its-1.2.0.gem (100%)
Fetching: multi_json-1.12.1.gem (100%)
Fetching: net-ssh-3.2.0.gem (100%)
Fetching: net-scp-1.2.1.gem (100%)
Fetching: net-telnet-0.1.1.gem (100%)
Fetching: sfl-2.2.gem (100%)
Fetching: specinfra-2.60.3.gem (100%)
Fetching: serverspec-2.36.0.gem (100%)
-----> serverspec installed (version 2.36.0)
       /opt/chef/embedded/bin/ruby -I/tmp/busser/suites/serverspec -I/tmp/busser/gems/gems/rspec-support-3.5.0/lib:/tmp/busser/gems/gems/rspec-core-3.5.2/lib /opt/chef/embedded/bin/rspec --pattern /tmp/busser/suites/serverspec/\*\*/\*_spec.rb --color --format documentation --default-path /tmp/busser/suites/serverspec
       
       Service "postgresql"
         should be enabled
         should be running
       
       Port "5432"
         should be listening
       
       File "/srv/my_tablespace"
         should be directory
         should be mode 700
         should be owned by "postgres"
         should be grouped into "postgres"
       
       Command "su - postgres -c 'psql -qtc "\l+ db2"'"
         stdout
           should match /db2.*localUser.*UTF8.*en_US\.UTF-8.*en_US\.UTF-8.*my_space/
       
       Finished in 0.28184 seconds (files took 0.29857 seconds to load)
       8 examples, 0 failures
       
       Finished verifying <default-ubuntu-1404> (0m10.01s).
```

### CODE REVIEW

**Comment by gravyboat** [comment]
This is going to cause some compatibility issues so I would like a review by someone else, other than that it looks good to me.

File: postgres/init.sls [hunk 1]
```diff
@@ -252,7 +272,7 @@ postgresql-ext-{{ ext_name }}-for-db-{{ name }}:
       - postgres_user: postgresql-user-{{ ext.user }}
     {% endif %}
     {% if ext.get('schema') %}
-      - postgres_schema: postgresql-schema-{{ ext.schema }}
+      - postgres_schema: postgresql-schema-{{ ext.schema }}-for-db-{{ name }}
```
Comment (MrMarvin) [inline postgres/init.sls]: This used to reference the same state at https://github.com/saltstack-formulas/postgres-formula/blob/master/postgres/init.sls#L192 , however the state id was simply wrong and resulted in a failing state call.

File: postgres/init.sls [hunk 2]
```diff
@@ -48,9 +48,11 @@ postgresql-running:
       - cmd: postgresql-cluster-prepared
 
 {% if postgres.pkgs_extra %}
-postgresql-extra-pkgs-installed:
+{% for pkg in postgres.pkgs_extra %}
```
Comment (MrMarvin) [inline postgres/init.sls]: Unsure about this one. According to our existing pillar data as well as that the defaults.yml does not actually provide an example, it is to assume that 'pkgs_extra' is expected to be a list. Therefore the loop to install each package individually is a safe way to handle this.

File: test/integration/default/serverspec/postgres_spec.rb [hunk 3]
```diff
@@ -16,6 +18,6 @@
   it { should be_grouped_into 'postgres' }
 end
 
-describe command('su - postgres -c "psql -qtc \"\l+ db2\""') do
-        its(:stdout) { should match(/db2.*localUser.*UTF8.*C.UTF-8.*C.UTF-8.*my_space/) }
+describe command(%q{su - postgres -c 'psql -qtc "\l+ db2"'}) do
```
Comment (MrMarvin) [inline test/integration/default/serverspec/postgres_spec.rb]: Proper string quote handling in ruby, yeay!

File: test/integration/default/serverspec/postgres_spec.rb [hunk 4]
```diff
@@ -16,6 +18,6 @@
...
+        its(:stdout) { should match(/db2.*localUser.*UTF8.*en_US\.UTF-8.*en_US\.UTF-8.*my_space/) }
```
Comment (MrMarvin) [inline test/integration/default/serverspec/postgres_spec.rb]: 90cfea1e15006c9781a3c4d2984a0f852b62324c changed the pillar.example data without caring for the spec... 😒

File: postgres/init.sls [hunk 5]
```diff
@@ -132,8 +134,17 @@ postgresql-tablespace-{{ name }}:
     - name: {{ name }}
     - directory: {{ tblspace.directory }}
     - user: {{ tblspace.get('runas', postgres.user) }}
-{% if tblspace.get('user') %}
-    - db_user: {{ tblspace.user }}
+{% if tblspace.get('db_user') %}
+    - db_user: {{ tblspace.db_user }}
+{% endif %}
+{% if tblspace.get('db_password') %}
```
Comment (MrMarvin) [inline postgres/init.sls]: Nice additions. Specifying a connections user without password, hosts and the like doesnt make much sense...

File: postgres/init.sls [hunk 6]
```diff
@@ -172,8 +183,17 @@ postgresql-db-{{ name }}:
     - owner: {{ db.owner }}
     {% endif %}
     - user: {{ db.get('runas', postgres.user) }}
-    {% if db.get('user') %}
-    - db_user: {{ db.user }}
+    {% if db.get('db_user') %}
+    - db_user: {{ db.db_user }}
+    {% endif %}
+    {% if db.get('db_password') %}
```
Comment (MrMarvin) [inline postgres/init.sls]: Nice additions. Specifying a connections user without password, hosts and the like doesnt make much sense...

**Comment by MrMarvin** [comment]
What compatibility issues to you see here, @gravyboat? Looks good to me too.

**Comment by gravyboat** [comment]
@MrMarvin It's the format changes to the pillar content which makes it backwards incompatible. Otherwise yeah I agree that it looks good. Since no one else has commented I'm going to merge it.

**Pull request merged**